baseline-network-linux-hld.md

```diff
@@ -0,0 +1,163 @@
+# Unified Open Networking Hardware Support
+
+[[_TOC_]]
+
+## Scope
+
+The current open network operating system (NOS) market is fragmented and has
+multiple competing projects with different focus, architecture and level of
+maturity. Although given those differences, all of the currently available
+and open network operating systems are utilising a mainline Linux kernel and
+need to build and maintain platform specific abstractions (e.g out of tree
+drivers) for peripheral switch hardware.
+
+This proposal aims to reduce the fragmentation by providing a common base for
+networking hardware support, without addressing the question of how to abstract
+and control the switch forwarding plane itself.
+
+The intention of this project is not to replace or compete with any of the
+existing network operating systems, but rather to provide a common and
+extensible layer on which those can be built. Its principles and paradigms are
+largely aligned and adapted from the
+[OCP OpenNetworkLinux Project](https://github.com/opencomputeproject/OpenNetworkLinux).
+This proposal specifically seeks an agreement between the developers and
+maintainers of the aforementioned network operating system communities, to be
+able to shared common implementations to support open networking hardware.
+
+Support for open networking hardware platforms is currently distributed over
+various different projects. Many platforms, especially older ones are supported
+by OpenNetworkLinux, many are supported by SONiC, and some are supported by
+DentOS, which is based on OpenNetworkLinux, but targets different devices. The
+open hardware support is usually very tightly integrated in the build systems,
+which makes sharing and reusing the support hard.
+
+## Requirements and Dependencies
+
+## Functional Description
+
+### Base Open Networking Hardware Support
+
+The base networking hardware support combines hardware support from multiple
+sources, and therefore will need to support multiple kinds of APIs.
+
+Possible APIs are:
+
+- SONiC PDDF, due to the large amount of available platforms
+  https://github.com/sonic-net/SONiC/blob/master/doc/platform/brcm_pdk_pddf.md
+- S3IP as described in https://github.com/sonic-net/SONiC/pull/1068
+- ONLP, as used by ONL and DentOS
```

Comment:
For SONiC, a reason is given (due to the large amount of available platforms). It would seem logical to add rationales for the other two APIs as well.

Reply:
ONLP supports approximately 170 platforms currently. SONiC shows 122 but some may not work in the current release.  In general both support about the same number of ~ < 5 year old platforms.

Reply:
SONiC is similar to ONLP. Just because a vendor says Platform X supports SONiC, it does not mean the Platform is currently supported. However, I would agree, SONiC's list ofsupported platforms is probably more up to date.